src/deck.rs

```diff
@@ -27,12 +27,24 @@ pub fn get_keys() -> (RsaPrivateKey, RsaPublicKey, ThreadRng) {
 
 impl Deck {
     pub fn new(domain: &str, plaintext: &str) -> Deck {
-        let domain = domain.to_string();
+        let mut domain = domain.to_string();
+        let vault_dir = vault_path();
+        let mut counter = 1;
+
+        while Self::domain_exists(&vault_dir, &domain) {
+            domain = format!("{}_{}", domain, counter);
+            counter += 1;
+        }
```

Comment:
the ideology here, I love it, such a thoughtful way to solve the problem
you're a gem💎